apps/condo/domains/meter/tasks/sendVerificationDateReminder.js

```diff
@@ -0,0 +1,233 @@
+                    // 2. Filter out messages related to other meters
+                    // 3. Filter out messages sent before current reminder window
+                    const messageCreateAt = dayjs(sentMessage.createdAt)
+                    const endOfReminderWindow = dayjs(meter.nextVerificationDate)
+                    const startOfReminderWindow = endOfReminderWindow.add(-reminderWindowSize, 'day')
+
+                    return sentMessage.user.id === resident.user.id
+                        && messageCreateAt.unix() >= startOfReminderWindow.unix()
+                })
+                return message == null
+            })
+
+            // create a pair again
+            // and filter out meter with empty residents
+            return {
+                meter,
+                residents: residentsWithoutNotifications,
+            }
+        })
+        .filter(pair => pair.residents.length > 0)
+}
+
+const getOrganizationLang = async (context, id) => {
+    const organization = await Organization.getOne(context, { id, deletedAt: null })
+
+    /**
+     * Detect message language
+     * Use DEFAULT_LOCALE if organization.country is unknown
+     * (not defined within @condo/domains/common/constants/countries)
+     */
+    return get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+}
+const generateReminderMessages = async ({ context, reminderWindowSize, reminders }) => {
+    const messages = []
+
+    await Promise.all(reminders.map(async (reminder) => {
+        const { meter, residents } = reminder
+        const lang = await getOrganizationLang(context, meter.organization.id)
+
+        // prepare a message for each resident
+        messages.push(
+            ...residents.map(resident => ({
+                sender: { dv: 1, fingerprint: 'meters-validation-date-cron-push' },
+                to: { user: { id: resident.user.id } },
+                type: METER_VERIFICATION_DATE_REMINDER_TYPE,
+                lang,
+                meta: {
+                    dv: 1,
+                    data: {
+                        reminderWindowSize,
+                        reminderDate: dayjs(meter.nextVerificationDate).locale(lang).format('D MMM'),
+                        meterId: meter.id,
+                    },
+                },
+            }))
+        )
+    }))
+
+    return messages
+}
+
+const sendReminders = async ({ context, messages }) => {
+    await Promise.all(messages.map(async (message) => {
+        await sendMessage(context, message)
+    }))
+}
+
+const sendVerificationDateReminder = async ({ date, searchWindowDaysShift, daysCount }) => {
+    // prepare vars
+    if (!date) date = dayjs().format('YYYY-MM-DD')
+    const reminderWindowSize = searchWindowDaysShift + daysCount
+
+    // initialize context stuff
+    const { keystone } = await getSchemaCtx('Meter')
+    const context = await keystone.createContext({ skipAccessControl: true })
```

Comment:
you can use keystone as context. you don't need to create a new one

Reply:
Removed